### PR TITLE
add input formatters utils (#322)

### PR DESCRIPTION
**Pull request by imGok**
inputut

Fix aswell the red field when error msg is null

### CODE REVIEW

**Review by huynguyennovem — COMMENTED** [review]
Looks good! Pls apply for My Space and Shared Space Document too!

File: lib/presentation/util/helper/input_formatters_utils.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'package:flutter/services.dart';
+
+class InputFormattersUtils {
+  static final specialCharactersRegex = FilteringTextInputFormatter.allow(RegExp('[0-9a-zA-Z]'));
```
Comment (hoangdat) [inline lib/presentation/util/helper/input_formatters_utils.dart:36]: We should support `-` `_` `.``[``]`.

Comment (hoangdat) [reply lib/presentation/util/helper/input_formatters_utils.dart:36]: And we can prevent the emotion icon?

Comment (imGok) [reply lib/presentation/util/helper/input_formatters_utils.dart:36]: emotion ?

File: lib/presentation/view/modal_sheets/edit_text_modal_sheet_builder.dart [hunk 2]
```diff
@@ -166,6 +174,8 @@ class EditTextModalSheetBuilder {
                       Padding(
                           padding: EdgeInsets.only(top: 20),
                           child: TextFormField(
+                            keyboardType: TextInputType.visiblePassword,
```
Comment (imGok) [inline lib/presentation/view/modal_sheets/edit_text_modal_sheet_builder.dart:177]: this is to hide emoji button in keyboard

File: lib/presentation/util/helper/input_formatters_utils.dart [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'package:flutter/services.dart';
+
+class InputFormattersUtils {
+  static final noSpecialCharactersRegex = FilteringTextInputFormatter.allow(RegExp('[0-9a-zA-Z-_.\\[\\]]', dotAll: false));
```
Comment (imGok) [inline lib/presentation/util/helper/input_formatters_utils.dart:36]: dotAll avoids "." from taking all characters

**Comment by imGok** [comment]
@hoangdat done!

**Review by hoangdat — APPROVED** [review]

**Review by huynguyennovem — APPROVED** [review]

**Pull request merged**